scripts/check_pr_scope.py

```
@@ -0,0 +1,202 @@
+    import re
+    pattern = r'^(feat|fix|chore|refactor|docs|test)/[a-z-]+(-[a-z-]+)*$'
+    if not re.match(pattern, branch_name):
+        print("❌ Branch name must follow pattern: type/short-description")
+        print(f"   Current: {branch_name}")
+        print("   Examples: feat/add-user-auth, fix/validate-input, chore/update-deps")
+        return False
+
+    return True
+
+
+def main():
```

Pylint (3.3.8)

[refactor] Too many branches (23/12)

(R0912)